extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/grpc/GrpcTracingServerInterceptor.java

```diff
@@ -147,7 +145,8 @@ public void onComplete() {
 
         @Override
         public void onReady() {
-            try {
+            Scope scope = spanContext.makeCurrent();
+            try (scope) {
```

Comment:
Can you check that `scope.close()` (which will be called automatically) is non-blocking?

Reply:
It is the same implementation we use for Vert.x, provided by https://github.com/quarkusio/quarkus/blob/main/extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/QuarkusContextStorage.java#L51-L83